thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/FetchAnomaliesInRangeAndOutputCSV.java

```diff
@@ -0,0 +1,193 @@
+    mergedAnomalyResultDAO = DaoProviderUtil
+        .getInstance(com.linkedin.thirdeye.datalayer.bao.jdbc.MergedAnomalyResultManagerImpl.class);
+  }
+
+  public static void outputResultNodesToFile(File outputFile, List<FetchMetricDataAndExistingAnomaliesTool.ResultNode> resultNodes)
+    throws IOException{
+    BufferedWriter bw = new BufferedWriter(new FileWriter(outputFile));
+
+    bw.write(String.join(",", resultNodes.get(0).getSchema()));
+    bw.newLine();
+    for (FetchMetricDataAndExistingAnomaliesTool.ResultNode n : resultNodes){
+      bw.write(n.toString());
+      bw.newLine();
+    }
+    bw.close();
+  }
+  /**
+   * Ouput merged anomaly results for given metric and time range
+   * @param args List of arguments
+   *             0: path to persistence file
+   *             1: collection name
+   *             2: metric name
+   *             3: start time in ISO format
+   *             4: end time in ISO format
+   *             5: (optional): Output path
+   */
+  public static void main(String args[]){
+    if(args.length < 5){
+      System.out.println("Insufficient number of arguments");
+      return;
+    }
+
+    FetchMetricDataAndExistingAnomaliesTool thirdEyeDAO = null;
+    try {
+      thirdEyeDAO = new FetchMetricDataAndExistingAnomaliesTool(new File(args[0]));
+    }
+    catch (Exception e){
+      System.out.println(e.getMessage());
+      return;
+    }
+
+    String collection = args[1];
+    String metric = args[2];
+    String output_folder = DEFAULT_OUTPUT_FOLDER;
+    DateTime startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(args[3]);
+    DateTime endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(args[4]);
+    DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd");
+    if(args.length >= 6 && (new File(args[5])).exists()){
+      output_folder = args[5];
+    }
+
+
+
+    // Print Merged Results
+    List<FetchMetricDataAndExistingAnomaliesTool.ResultNode> resultNodes = thirdEyeDAO.fetchMergedAnomaliesInRange(
+        collection, metric, args[3], args[4]);
+
+    System.out.println("Printing merged anaomaly results from db...");
+    String outputname = output_folder +
+        "merged_" + args[1] + "_" + fmt.print(startTime) + "_" + fmt.print(endTime) + ".csv";
+    try {
+      outputResultNodesToFile(new File(outputname), resultNodes);
+    }
+    catch (IOException e){
+      System.out.println(e.getMessage());
+    }
+    System.out.println("Finish job of printing merged anaomaly results from db...");
+
+
+    resultNodes.clear();
+    // Print Raw Results
+    resultNodes = thirdEyeDAO.fetchRawAnomaliesInRange(collection, metric, args[3], args[4]);
+
+    System.out.println("Printing raw anaomaly results from db...");
+    outputname = output_folder +
+        "raw_" + args[1] + "_" + fmt.print(startTime) + "_" + fmt.print(endTime) + ".csv";
+    try {
+      outputResultNodesToFile(new File(outputname), resultNodes);
+    }
```

Comment:
handle this exception in the method instead of handling it in the main everytime it is called